### PR TITLE
Bump go version to 1.16

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
This patch updates:
- Go to 1.16.3
- Run `go mod tidy` 

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -12,6 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: k8s.gcr.io/nfd/node-feature-discovery-operator
-  newTag: v0.2.0
-
+  newName: quay.io/eduardoarango/node-feature-discovery-operator
+  newTag: v0.2.0-22-ge57fe01f-dirty
```
Comment (marquiz) [inline config/manager/kustomization.yaml:16]: Don't change these, please

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ spec:
             capabilities:
               drop: ["ALL"]
           command:
-            - /manager
+            - /node-feature-discovery-operator
```
Comment (marquiz) [inline config/manager/manager.yaml:33]: Unrelated

**Review by marquiz — CHANGES_REQUESTED** [review]
Remember to also submit a accompanying PR against https://github.com/kubernetes/test-infra

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (ArangoGutierrez) [reply config/manager/kustomization.yaml:16]: damm kustomize... sorry

**Comment by ArangoGutierrez** [comment]
> Remember to also submit a accompanying PR against https://github.com/kubernetes/test-infra

working on that

**Comment by ArangoGutierrez** [comment]
https://github.com/kubernetes/test-infra/pull/21726

File: config/manager/manager.yaml (hunk 2 above)
Comment (ArangoGutierrez) [reply config/manager/manager.yaml:33]: this is fixed on #53 , had to use it here to.
after merging #53 this line won't be needed

**Review by marquiz — COMMENTED** [review]
Sorry to nag about the commit message as it's not aligned with the content. Could you update it, e.g. to something like the PR title `Bump go version and tidy up go.sum`

**Comment by ArangoGutierrez** [comment]
> Bump go version and tidy up go.sum

done, now commit has a good and descriptive message

**Comment by marquiz** [comment]
> commit has a good and descriptive message

Don't see any change there 🤔

**Comment by marquiz** [comment]
> Don't see any change there 🤔

Now I see it. I just think that subject of the PR ("Bump go version") would be far better subject to the commit message than the current "update dependencies". Which is quite misleading as this patch does not really udpate any dependencies

**Comment by ArangoGutierrez** [comment]
> > Don't see any change there thinking
> 
> Now I see it. I just think that subject of the PR ("Bump go version") would be far better subject to the commit message than the current "update dependencies". Which is quite misleading as this patch does not really udpate any dependencies

got it

**Review by marquiz — APPROVED** [review]
Thanks. Now we're golang 1.16 everywhere
/lgtm
